packages/server/modules/workspaces/graph/resolvers/workspaces.ts

```diff
@@ -175,11 +171,41 @@ export = FF_WORKSPACES_MODULE_ENABLED
 
           return workspace
         },
-        updateRole: async () => {
-          throw new WorkspacesNotYetImplementedError()
-        },
-        deleteRole: async () => {
-          throw new WorkspacesNotYetImplementedError()
+        updateRole: async (_parent, args, context) => {
+          const { userId, workspaceId, role } = args.input
+
+          authorizeResolver(context.userId, workspaceId, Roles.Workspace.Admin, null)
```

Comment:
I'd still put this in the service layer, to be sure that this check always happens regardless of where you trigger it from. IMO its fine that it can't be mocked out/injected. That said, you can leave it as is for now.

An important fix that has to happen, however, is that you should not ever leave the 4th param of `authorizeResolver` empty, if possible. This is where those resource access rules from the GQL `context` should be injected to validate that the token has appropriate rights.

Make sure this is done in all relevant resolvers in this PR

Reply:
sorry sorry, this had been explained and I forgot. will keep in mind 👍

will move `authorizeResolver` in a follow-up, ty for letting this through